### PR TITLE
Skip custom domain configurations when building ID token issuer for SaaS apps

### PR DESCRIPTION
**Pull request by Yasasr1**
### Proposed changes in this pull request
$subject

### When should this PR be merged
This pr should be merged after the following framework pr is merged
- https://github.com/wso2/carbon-identity-framework/pull/5700

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 1]
```diff
@@ -4157,6 +4162,26 @@ public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2
...
+    public static String getIdTokenIssuer(String tenantDomain, String clientId) throws IdentityOAuth2Exception {
+
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            try {
+                ServiceURLBuilder serviceURLBuilder = ServiceURLBuilder.create();
+                if (SAAS_APP_IDS.contains(clientId)) {
+                    serviceURLBuilder.setSkipCustomDomain(true);
+                }
+
+                return serviceURLBuilder.addPath(OAUTH2_TOKEN_EP_URL).build().getAbsolutePublicURL();
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:4174]: Alternative way to add the logic.

 return ServiceURLBuilder.create().addPath(OAUTH2_TOKEN_EP_URL). setSkipCustomDomain(SAAS_APP_IDS.contains(clientId)).build().getAbsolutePublicURL();

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 2]
```diff
@@ -383,6 +384,10 @@ public class OAuth2Util {
 
     private static final String BASIC_AUTHORIZATION_PREFIX = "Basic ";
 
+    private static final List<String> SAAS_APP_IDS = Arrays.asList(
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:387]:
```suggestion
    private static final List<String> PORTAL_APP_IDS = Arrays.asList(
```
Console & MyAccount are not SaaS apps. In some places those apps are identified as portal apps. Hence we can have that or any convenient name.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java (hunk 1 above)
Comment (Yasasr1) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: Updated with [5fc4333](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2458/commits/5fc43339d35766c8996f2ff612eeda9ea75f37e7)

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java (hunk 2 above)
Comment (Yasasr1) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: Updated with [5fc4333](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2458/commits/5fc43339d35766c8996f2ff612eeda9ea75f37e7)

**Review by sadilchamishka — APPROVED** [review]

**Pull request merged**